### ISSUE TITLE
Certain attributes behave differently given the same value

### ISSUE THREAD

**Issue by bernardmcmanus** [hyperapp/hyperapp#628]
__Environment:__
* `OSX 10.11.6`
* `Chrome 62.0.3202.94`
* `hyperapp@1.1.2`

Consider the following component:
```jsx
const Input = () =>
  <input type="text" spellcheck="false" aria-invalid="false" />;
```

I would expect this to be rendered as:
```html
<input type="text" spellcheck="false" aria-invalid="false" />
```

But the actual output is:
```html
<input type="text" spellcheck="true" aria-invalid="false" />
```

I'm able get the desired `spellcheck` value by setting it to an empty string in the component, but `aria-invalid` behaves differently
```jsx
const Input = () =>
  <input type="text" spellcheck="" aria-invalid="" />;
```

Now the rendered output is:
```html
<input type="text" spellcheck="false" aria-invalid />
```

Ultimately, I have to pass different values to the `spellcheck` and `aria-invalid` attributes in the component to get the same values in the output:
```jsx
const Input = () =>
  <input type="text" spellcheck="" aria-invalid="false" />;
```
```html
<input type="text" spellcheck="false" aria-invalid="false" />
```

**Comment by bernardmcmanus** [comment]
I may have some time to dig into this and submit a fix later this week - just submitting an issue to track.

**Comment by jorgebucaran** [comment]
@bernardmcmanus Yeah, you want to look at [these lines](https://github.com/hyperapp/hyperapp/blob/a6eec2b73794208da0e6a496e7d68c7ce7fa350f/src/index.js#L145-L165).

By the way, I just checked against Preact and got the same result as in Hyperapp.

**Comment by SkaterDad** [comment]
Seems like a quirk in how browsers treat DOM node properties?

**Comment by bernardmcmanus** [comment]
@SkaterDad I'm sure there are some cross browser quirks, but this particular case works fine with vanilla JS:
```js
var input = document.createElement('input');
input.type = 'text';
document.body.appendChild(input);
input.setAttribute('spellcheck', false);
input.setAttribute('aria-invalid', false);
```
```html
<input type="text" spellcheck="false" aria-invalid="false">
```
Same result if I pass `'false'` strings as the attribute values.

**Comment by SkaterDad** [comment]
Your vanilla version isn't quite the same.  Hyperapp only uses setAttribute when the property doesn't exist on the DOM node.  In other cases, it directly sets the property.

**Comment by bernardmcmanus** [comment]
Agreed - my point is the browser is capable of properly setting those attribute values regardless of type as long as you call the setter. The trick is going to be how to decide when to assign directly.

**Comment by bernardmcmanus** [comment]
> The trick is going to be how to decide when to assign directly.

Yikes 😬 easier said than done.

https://github.com/facebook/react/blob/2cf9063318014dcc9a4839bb15ebb89a55198fce/packages/react-dom/src/shared/DOMProperty.js

**Comment by jorgebucaran** [comment]
Yep, I'm on the "this is not a big deal" camp. I prefer keeping core simple and small even if it means sacrificing some correctness for edge cases like this one as long as I can work around it in userland without too much trouble.

**Comment by bernardmcmanus** [comment]
That's fair. I'll see if I can't come up with something light that at least covers most cases. I agree this certainly doesn't warrant a bloated lookup helper.

**Comment by jorgebucaran** [comment]
@bernardmcmanus Thanks! See also #629.

**Comment by osdevisnot** [comment]
Maybe I am too new to this, but can someone please shade some light on why we try to set the dom property directly ``` element[name] = value == null ? '' : value ``` as opposed to using ``` element.setAttribute(name, value) ``` always?

**Comment by jorgebucaran** [comment]
@osdevisnot To be as accurate as possible when manipulating the DOM. It used to be much simpler than what we currently have, but also less correct. For example, we want to call setAttribute and removeAttribute for SVG elements but want to set event handlers, classNames, innerHTML, etc., as properties.

**Comment by jorgebucaran** [comment]
Can anyone summarize what the problem is here? What's the behavior we want? Maybe we can compromise a bit (punt-intended) byte-wise.

**Comment by frenzzy** [comment]
@jorgebucaran the problem is unexpected hyperapp behavior:
```js
const view = () => <input spellcheck="false" /> // renders =>
// actual:   <input spellcheck="true" />
// expected: <input spellcheck="false" />
```
fixed here: #629
